### ISSUE TITLE
Pointer not set for location referenced in il_getParmStr()

### ISSUE THREAD

**Issue by tools400** [sitemule/ILEastic#38]
Hi,

Procedure il_getParmStr() issues a "Pointer not set for location referenced." error, when no query parameters have been specified with the request URL.

```
   124      PSLIST pParmList = pRequest->parmList;                  
   125                                                              
   126      for (pNode = pParmList->pHead; pNode; pNode=pNode->pNext) {
   127         PSLISTKEYVAL parm = pNode->payloadData;                   
   128         if (keyLen == parm->key.Length) { 
```

Line 126: Pointer not set for location referenced, because pParmList is null.

It works fine, when at least the question mark has been added at the end of the URL.
You can easily reproduce the problem with the "PLUGIN" example.
I am not sure whether or not that is a bug or works as designed. But since il_getRequestHeader() works more gentle, I would expect il_getParmStr() to behave the same.

Regards,

Thomas.

**Comment by m1h43l** [comment]
Hi Thomas,

this is a know issue , see #29 (though the scope of the error is not correct expressed in the title), and there is a unit test which covers this already.

Best regards

Mihael

**Comment by m1h43l** [comment]
This has been fixed with the latest commit  c9c6d91
